tests/test_clients.py

```diff
@@ -112,11 +114,44 @@ async def test_title_search(paper_attributes: dict[str, str]) -> None:
             ),
         )
         details = await client.query(title=paper_attributes["title"])
+
+        # matches the citation pattern, not the specific citation count
+        # search because the count will be somewhere in the middle of the string
+        expected_citation_format = re.search(
+            DocDetails.CITATION_COUNT_REGEX_PATTERN,
+            paper_attributes["formatted_citation"],
+        )
+        actual_citation_format = re.search(
+            DocDetails.CITATION_COUNT_REGEX_PATTERN,
+            details.formatted_citation,  # type: ignore[union-attr]
+        )
+        assert (
+            expected_citation_format is not None
+        ), "Expected string should match the citation pattern"
+        assert (
+            actual_citation_format is not None
+        ), "Actual string should match the citation pattern"
+
+        expected_remaining = (
+            paper_attributes["formatted_citation"][: expected_citation_format.start()]
+            + paper_attributes["formatted_citation"][expected_citation_format.end() :]
+        )
+
+        actual_remaining = (
+            details.formatted_citation[: actual_citation_format.start()]  # type: ignore[union-attr]
+            + details.formatted_citation[actual_citation_format.end() :]  # type: ignore[union-attr]
+        )
+
+        # Assert that the parts of the strings outside the citation count are identical
+        assert (
+            expected_remaining == actual_remaining
+        ), "Formatted citation text should match except for citation count"
```

Comment:
So we are doing a regex search to find a region, then removing that region, and lastly doing an equality check on the front/back end.

I think it would be simpler to do one regex removal of the region, and directly equality compare the remanants.

I also think this is what you originally had 😅 😓  hahaha sorry! I actually realized I misunderstood your original logic last night, my bad here.

What do you think?

Reply:
definitely! no worries, communicating through PRs can be hard sometimes. Thank you!! adjusting now.

before I merge this I'm also going to look into why the cassette file aren't (hasn't) been updated